internal/csi-addons/rbd/reclaimspace.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2021 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+import (
+	"context"
+	"fmt"
+
+	csicommon "github.com/ceph/ceph-csi/internal/csi-common"
+	rbdutil "github.com/ceph/ceph-csi/internal/rbd"
+	"github.com/ceph/ceph-csi/internal/util"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	rs "github.com/csi-addons/spec/lib/go/reclaimspace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// ReclaimSpaceControllerServer struct of rbd CSI driver with supported methods
+// of CSI-addons reclaimspace controller service spec.
+type ReclaimSpaceControllerServer struct {
+	*rs.UnimplementedReclaimSpaceControllerServer
+}
+
+// NewReclaimSpaceControllerServer creates a new IdentityServer which handles
+// the Identity Service requests from the CSI-Addons specification.
+func NewReclaimSpaceControllerServer() *ReclaimSpaceControllerServer {
+	return &ReclaimSpaceControllerServer{}
+}
+
+func (rscs *ReclaimSpaceControllerServer) RegisterService(server grpc.ServiceRegistrar) {
```

Comment:
as it is exported good to add a comment about this function too. @nixpanic

Reply:
It is the implementation of the `CSIAddonsService`, documentation can be found there.